CSharp/JobPrep/Program.cs

```diff
@@ -0,0 +1,333 @@
+                    {
+                        Console.WriteLine("Using existing pool {0}", poolName);
+                        found = true;
+
+                        if (!p.ListVMs().Any<IVM>())
+                        {
+                            Console.WriteLine("There are no VMs in this pool. No tasks will be run until at least one VM has been added via resizing.");
+                            Console.WriteLine("Resizing pool to add 3 VMs. This might take a while...");
+                            p.Resize(3);
+                        }
+
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    Console.WriteLine("Creating pool: {0}", poolName);
+                    // if pool not found, call CreatePool
+                    //You can learn more about os families and versions at:
+                    //http://msdn.microsoft.com/en-us/library/azure/ee924680.aspx
+                    ICloudPool pool = pm.CreatePool(poolName, targetDedicated: 20, vmSize: "small", osFamily: "3");
+                    pool.Commit();
+                }
+            }
+        }
+
+        private static void ListPools(IBatchClient client)
+        {
+            using (IPoolManager pm = client.OpenPoolManager())
+            {
+                Console.WriteLine("Listing Pools\n=============");
+                // Using optional select clause to return only the name and state.
+                IEnumerable<ICloudPool> pools = pm.ListPools(new ODATADetailLevel(selectClause: "name,state"));
+                foreach (var p in pools)
+                {
+                    Console.WriteLine("pool " + p.Name + " is " + p.State);
+                }
+                Console.WriteLine();
+            }
+        }
+
+        private static void ListWorkItems(IBatchClient client)
+        {
+            // All Workitem, Job, and Task related operation start from WorkItemManager
+            using (IWorkItemManager wm = client.OpenWorkItemManager())
+            {
+                Console.WriteLine("Listing Workitems\n=================");
+                IEnumerable<ICloudWorkItem> wis = wm.ListWorkItems();
+                foreach (var w in wis)
+                {
+                    Console.WriteLine("Workitem: " + w.Name + " State:" + w.State);
+                }
+                Console.WriteLine();
+            }
+        }
+
+        private static async Task<bool> PollJobPrepTaskAsync(IPoolManager pm, string pool, string vm, string workitem, string job, int sleep)
+        {
+            while(true)
+            {
+                try
+                {
+                    Console.WriteLine("polling prep task on " + vm);
+                    var preptask = await pm.GetJobPreparationTaskStatusAsync(pool, vm, workitem, job);
+                    if (preptask.State == JobPreparationTaskState.Completed)
+                        return preptask.ExitCode == 0;
```

Comment:
minor: I personally find if statements without { }'s to be more confusing than they are worth.  Add { } to this?